src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -42,7 +42,6 @@ using namespace eprosima::fastrtps::rtps;
 
 StatefulReader::~StatefulReader()
 {
-    std::lock_guard<std::recursive_timed_mutex> guard(mp_mutex);
```

Comment:
This was added to fix a race condition and you are reverting it here, right?

Reply:
Yes!!